### PR TITLE
fix(api-graphql): server side iam auth mode is ineffective

### PR DESCRIPTION
**Pull request by HuiSF**
#### Description of changes

1. Changed the api-rest exported internal `post` function for api-graphql to make it server side usable (can inject the instance of `AmplifyClassV6` from the server context)
2. Changed the api-graphql server `graphql` method `wrapper` implementation to ensure it's an `await` -able function so Amplify server context can ensure destroying context only after this function has completed
3. Updated test suites

#### Issue #, if available

https://github.com/aws-amplify/amplify-js/issues/12984
https://github.com/aws-amplify/amplify-js/issues/12931
https://github.com/aws-amplify/amplify-js/issues/12867

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/api-rest/src/internals/index.ts
Comment (HuiSF) [inline packages/api-rest/src/internals/index.ts:1]: @AllanZhengYP Please review this, ty!

Comment (AllanZhengYP) [reply packages/api-rest/src/internals/index.ts:1]: Thanks! What's the difference of this change v.s. using the server version of internal post? https://github.com/aws-amplify/amplify-js/blob/hui/fix/api-graphql/server-side-iam/packages/api-rest/src/internals/server.ts

Comment (HuiSF) [reply packages/api-rest/src/internals/index.ts:1]: GQL impl. doesn't pick the client vs server versions, it just passing the Amplify instances down into the post API.

**Review by AllanZhengYP — APPROVED** [review]
The internal POST handler part of change looks good to me! Thank you @HuiSF

**Review by svidgen — APPROVED** [review]

**Review by stocaaro — APPROVED** [review]

**Pull request merged**